receiver/prometheusoperatorreceiver/DESIGN.md

```diff
@@ -0,0 +1,142 @@
+## Design Goals
+
+### Vision
+Provide a nearly zero-config migration path from PrometheusOperator defined scraping configs.  
+
+### Scope
+Only scrape config relevant CRDs (CustomResourceDefinition) are in scope of this receiver.    
+As of this writing the relevant CRDs are: 
+- ServiceMonitor
+- PodMonitor
+
+The receiver should be able to query this CRDs from Kubernetes, attach watchers and reconfigure its scraping config 
+based on existing CRs (CustomResource) in Kubernetes.
+
+Changes as well as creations or deletions of said CRs should trigger reconciliation of the scrape config. 
+Active querying of the API should only occur during startup and in a defined interval of the receiver. 
+
+Namespace limitations should be possible for shared cluster scenarios.     
+This should be implemented using industry standard methods. 
+
+
+### Out of scope
+Excluded are other relevant CRDs such as: 
+- PrometheusRules
+- ThanosRuler
+- Probes
+- Alertmanagers
+- AlertmanagerConfigs
+
+This includes concepts like alerting and black box probing of defined targets. 
+
+
+## PrometheusOperator CR watcher
+The CR watcher is the component responsible for querying CRs from Kubernetes API and triggering a reconciliation.
+
+### Major components of CR watcher
+- **[ListenerFactory](https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/informers/monitoring.go):**
+  the component which creates listeners on the CR
+- **[APIClient](https://github.com/prometheus-operator/prometheus-operator/tree/main/pkg/client):**
+  PrometheusOperator API client
+
+## Config generator
+The config generator is triggered by the watcher and generates a Prometheus config as a byte array.   
+Instead of writing it onto a disk the configuration is unmarshalled using the Prometheus config loader, 
+which is already in use by the `prometheusreceiver` resulting in a full Prometheus config. 
+
+In case of an [Agent](#Collector vs Agent deployment) deployment, which is signaled with the `limit_to_node` option, 
+only local endpoints will be fetched, endpoints should be filtered so that only pods are scraped which are scheduled 
```

Comment:
Makes sense, but I'm not sure how to solve this.    
The only option other than introducing a new index in Kubernetes I see is the introduction of a shared cache. This could be maybe done as extension.

Reply:
I'm just noting it; I don't think it is easily solvable.  I think we should not recommend using a daemonset with servicemonitors to users because of this.